x-pack/plugins/endpoint/server/routes/alerts/details/lib/pagination.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { GetResponse, SearchResponse } from 'elasticsearch';
+import { RequestHandlerContext } from 'src/core/server';
+import { AlertData, AlertHits, Direction, Maybe } from '../../../../../common/types';
+import { EndpointConfigType } from '../../../../config';
+import { searchESForAlerts, Pagination } from '../../lib';
+import { AlertSearchQuery, SearchCursor, AlertDetailsRequestParams } from '../../types';
+import { BASE_ALERTS_ROUTE } from '../..';
+
+/**
+ * Pagination class for alert details.
+ */
+export class AlertDetailsPagination extends Pagination<
+  AlertDetailsRequestParams,
+  GetResponse<AlertData>
+> {
+  constructor(
+    config: EndpointConfigType,
+    requestContext: RequestHandlerContext,
+    state: AlertDetailsRequestParams,
+    data: GetResponse<AlertData>
+  ) {
+    super(config, requestContext, state, data);
+  }
+
+  protected async doSearch(
+    direction: Direction,
+    cursor: SearchCursor
+  ): Promise<SearchResponse<AlertData>> {
+    const reqData: AlertSearchQuery = {
+      pageSize: 1,
+      sort: this.config.alertResultListDefaultSort,
+      order: this.config.alertResultListDefaultOrder as Direction,
+    };
+
+    if (direction === Direction.asc) {
+      reqData.searchAfter = cursor;
+    } else {
+      reqData.searchBefore = cursor;
+    }
+
+    const response = await searchESForAlerts(
+      this.requestContext.core.elasticsearch.dataClient,
+      reqData
+    );
+    return response;
+  }
+
+  protected getUrlFromHits(hits: AlertHits): Maybe<string> {
```

Comment:
consider taking just a single `_id` from a hit. Consider moving the conditional logic up the call stack.